src/databricks/sql/client.py

```diff
@@ -222,8 +227,36 @@ def read(self) -> Optional[OAuthToken]:
         logger.info("Successfully opened session " + str(self.get_session_id_hex()))
         self._cursors = []  # type: List[Cursor]
 
-        self._suppress_inline_warning = "use_inline_params" in kwargs
-        self.use_inline_params = kwargs.get("use_inline_params", True)
+        self.use_inline_params = self._set_use_inline_params_with_warning(
+            kwargs.get("use_inline_params")
+        )
+
+    def _set_use_inline_params_with_warning(self, user_value):
+        """Valid values are True, False, and "silent"
+
+        False: Use native parameters
+        True: Use inline parameters and log a warning
+        "silent": Use inline parameters and don't log a warning
+        """
+
+        if user_value is None:
```

Comment:
declare as Option and set default in parameter?

Reply:
Fixed!